### PR TITLE
Add argo client

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hadim** [comment]
ping @conda-forge/help-go

File: recipes/argo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: argo
+
+  url: https://github.com/argoproj/argo/releases/download/v{{ version }}/argo-darwin-amd64  # [osx]
+  sha256: 15dd9013654ea2a786d7eafc6f05e51e994ec2e7e595d84c27113a5f69eee918  # [osx]
```
Comment (scopatz) [inline recipes/argo/meta.yaml:12]: Please build from source, rather than repackaging binaries

Comment (hadim) [reply recipes/argo/meta.yaml]: I am fine building from source but I can get the install files to go to PREFIX. I tried looking at other go libs without success. Could you help?

Comment (scopatz) [reply recipes/argo/meta.yaml]: I don't know much about go, sorry.  Hopefully @conda-forge/help-go can help.

Comment (hadim) [reply recipes/argo/meta.yaml]: I tried multiple things still without success. Are you sure we can't build from official binaries? Those binaries are coming from the official repository. Here is an example of a golang conda forge package using the same strategy: https://github.com/conda-forge/go-spatial-feedstock/blob/master/recipe/meta.yaml#L10

Comment (scopatz) [reply recipes/argo/meta.yaml]: Yes, they may link to libraries we don't have on the system.

Comment (hadim) [reply recipes/argo/meta.yaml]: Looking at the size of the binaries, they are probably all statically linked and self-sufficient.

Comment (scopatz) [reply recipes/argo/meta.yaml]: There is some weird cos6 vs cos7 issue with go that I have run into before

Comment (hadim) [reply recipes/argo/meta.yaml]: I finally get it to compile on Linux and osx. Can we merge this and we can figure out windows later?

Comment (hadim) [reply recipes/argo/meta.yaml]: Or maybe there is a way to call `build.sh` during Windows CI?

**Comment by isuruf** [comment]
This has the same issue as https://github.com/conda-forge/conda-forge.github.io/issues/1052

**Comment by hadim** [comment]
Besides the licensing issue, I am also running into technical issues building on Windows.

File: recipes/argo/meta.yaml (hunk 1 above)
Comment (sodre) [reply recipes/argo/meta.yaml]: Sorry, I just saw things. I'll help out

Comment (hadim) [reply recipes/argo/meta.yaml]: You're a hero @sodre !

**Comment by sodre** [comment]
> Why python works for win? Any specific reason or just "luck"?
I needed it on my machine, it yarn was throwing errors on my side.

**Comment by sodre** [comment]
> What's the usual practice for win, use .exe or not?

GO will append the .exe automatically on windows. There is nothing we need to do extra.

We don't need to explicitly call argo.exe because Windows knows to append it as well.

**Comment by hadim** [comment]
LGTM for me.

**Comment by isuruf** [comment]
There's still the license issue

**Comment by hadim** [comment]
But it does not seem to be specifically related to this package. There are already multiple Go packages in conda-forge.

See my comment on https://github.com/conda-forge/conda-forge.github.io/issues/1052#issuecomment-639582258

**Comment by isuruf** [comment]
> There are already multiple Go packages in conda-forge.

Those packages are in violation too.

**Comment by hadim** [comment]
Licenses are not included in the recipe as well as the script that I use to generate them.

**Comment by isuruf** [comment]
Can you use the script to generate it at package build time? This is so that if the dependencies changed, the new licenses are packaged too.

**Comment by hadim** [comment]
Can't we just rerun the script each time we bump a new version?

**Comment by isuruf** [comment]
Dependencies of dependencies might change from the time you made the PR and the build actually happens.

**Comment by hadim** [comment]
So why including them in the recipe?

File: recipes/argo/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+set -euf
+
+yarn --cwd ui install --frozen-lockfile --ignore-optional --non-interactive
+yarn --cwd ui build
+
+# This could be a conda-package as a build requirement
+# then we would not reed to remove it
+go get bou.ke/staticfiles
```
Comment (isuruf) [inline recipes/argo/build.sh:9]: What about licenses for this package?

File: recipes/argo/build_library_licenses.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export GOPATH=$(go env GOPATH)
+export LIBRARY_LICENCES_PATH="$RECIPE_DIR/library_licenses/"
+export TEMP_DIR=$(mktemp -d)
+export ARGO_PATH="$TEMP_DIR/argo"
+
+cd $TEMP_DIR
+git clone https://github.com/argoproj/argo.git
+cd argo/
+git checkout v2.8.1
```
Comment (isuruf) [inline recipes/argo/build_library_licenses.sh:16]: Can you use `$SRC_DIR`?

**Comment by hadim** [comment]
Sorry but feel free to close or not merge. I don't have more time to help here.

**Comment by isuruf** [comment]
> So why including them in the recipe?

It's not needed, if the script can generate all of them at package build time.

For Rust, there was no way to have a complete script like the go-license one. So, the licenses were generated using a script, and then files added manually for ones that the script failed. There was a runtime check added that all files are present.

File: recipes/argo/build.sh (hunk 2 above)
Comment (sodre) [reply recipes/argo/build.sh:9]: This is not needed, it is used during compile time and then discarded.

**Pull request closed without merging**